src/components/CharactersPage/CharacterForm/CharacterForm.js

```diff
@@ -107,6 +108,19 @@ const CharacterForm = ({ onClose }) => {
     reValidateMode: "onSubmit",
   })
 
+  useEffect(() => {
+    if (isEdit) {
+      const cachedCharacters = client.cache.extract()
+
+      if (!has(cachedCharacters, `Person:${params[1]}`)) {
+        router.push({
+          pathname: "/characters/[[...params]]",
+          query: undefined,
+        })
+      }
+    }
+  }, [isEdit, params])
```

Comment:
This part of the code is used to disable the user from fetching deleted character by going back in the browser, (as in go to previous page) which will initiate the query `GET_CHARACTER` inside the form